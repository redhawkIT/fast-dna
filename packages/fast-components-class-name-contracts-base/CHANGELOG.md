# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

<a name="1.1.0"></a>
# 1.1.0 (2018-05-09)


### Bug Fixes

* travis-CI build-break ([#336](https://github.com/Microsoft/fast-dna/issues/336)) ([bffbf5e](https://github.com/Microsoft/fast-dna/commit/bffbf5e))


### Features

* add hypertext component ([#210](https://github.com/Microsoft/fast-dna/issues/210)) ([9e363ff](https://github.com/Microsoft/fast-dna/commit/9e363ff))
* Forked Class name contracts so we can have one for Base and one for MSFT ([#262](https://github.com/Microsoft/fast-dna/issues/262)) ([a4c54c0](https://github.com/Microsoft/fast-dna/commit/a4c54c0))
* **checkbox:** add new component with styles ([#252](https://github.com/Microsoft/fast-dna/issues/252)) ([3ad3988](https://github.com/Microsoft/fast-dna/commit/3ad3988))
* **label:** add new component and msft styles ([#265](https://github.com/Microsoft/fast-dna/issues/265)) ([0328028](https://github.com/Microsoft/fast-dna/commit/0328028))
* update code coverage on travis ([#330](https://github.com/Microsoft/fast-dna/issues/330)) ([63ab4f4](https://github.com/Microsoft/fast-dna/commit/63ab4f4))
