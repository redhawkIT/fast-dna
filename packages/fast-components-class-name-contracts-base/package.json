{
  "name": "@microsoft/fast-components-class-name-contracts-base",
  "description": "Type descriptions for the classnames that any given component can expect as props.",
  "version": "1.4.0",
  "author": {
    "name": "Microsoft",
    "email": "fwq@microsoft.com"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/Microsoft/fast-dna.git"
  },
  "main": "./dist/index.js",
  "types": "./dist/index.d.ts",
  "scripts": {
    "clean:dist": "node ../../build/clean.js dist",
    "prepare": "npm run clean:dist && tsc -p ./tsconfig.json",
    "test": "npm run tslint",
    "tslint": "tslint -c ./tslint.json '**/*.ts{,x}'",
    "tslint:fix": "tslint -c ./tslint.json --fix '**/*.ts{,x}'"
  },
  "license": "MIT",
  "devDependencies": {
    "@microsoft/fast-tslint-rules": "^1.4.0",
    "tslint": "^5.9.1",
    "typescript": "^2.7.2"
  }
}
