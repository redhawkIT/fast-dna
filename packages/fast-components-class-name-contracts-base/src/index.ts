export * from "./breadcrumb";
export * from "./button";
export * from "./card";
export * from "./checkbox";
export * from "./context-menu";
export * from "./context-menu-item";
export * from "./dialog";
export * from "./divider";
export * from "./horizontal-overflow";
export * from "./hypertext";
export * from "./image";
export * from "./label";
export * from "./managed-classes";
export * from "./progress";
export * from "./radio";
export * from "./tabs";
export * from "./text-field";
export * from "./toggle";
export * from "./typography";
