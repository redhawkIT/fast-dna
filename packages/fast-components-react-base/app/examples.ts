import ButtonExamples from "../src/button/examples.data";
export { ButtonExamples };
