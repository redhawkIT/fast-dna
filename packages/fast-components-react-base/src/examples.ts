import * as React from "react";

export type IExampleData<T> = T;

export interface IGenericExample<T> {
    name: string;

    component: React.ComponentClass;

    data: Array<IExampleData<T>>;
}
