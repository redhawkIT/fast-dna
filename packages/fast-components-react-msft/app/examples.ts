import ButtonExample from "../src/button/examples.data";
export {ButtonExample};

import DividerExample from "../src/divider/examples.data";
export {DividerExample};

import ToggleExamples from "../src/toggle/examples.data";
export {ToggleExamples};
