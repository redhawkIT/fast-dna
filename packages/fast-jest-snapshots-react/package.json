{
  "name": "@microsoft/fast-jest-snapshots-react",
  "description": "A small utility for automating Jest snapshot creation of React components",
  "version": "1.6.0",
  "author": {
    "name": "Microsoft",
    "email": "fwq@microsoft.com"
  },
  "license": "MIT",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/Microsoft/fast-dna.git"
  },
  "private": true,
  "main": "./dist/index.js",
  "types": "./dist/index.d.ts",
  "scripts": {
    "prepare": "tsc -p ./tsconfig.json",
    "test": "npm run tslint",
    "start": "webpack-dev-server",
    "tslint": "tslint -c ./tslint.json '**/*.ts{,x}'",
    "tslint:fix": "tslint -c ./tslint.json --fix '**/*.ts{,x}'"
  },
  "devDependencies": {
    "@microsoft/fast-tslint-rules": "^1.6.0",
    "@types/jest": "^22.2.2",
    "jest": "^22.4.3",
    "react": "^16.3.0",
    "ts-jest": "^22.4.2",
    "tslint": "^5.9.1",
    "typescript": "^2.8.1"
  },
  "dependencies": {
    "@types/react": "^16.3.0",
    "react-test-renderer": "^16.3.0"
  },
  "peerDependencies": {
    "jest": "^22.0.0",
    "react": "^16.3.0"
  }
}
