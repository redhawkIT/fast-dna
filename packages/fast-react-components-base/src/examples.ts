import * as React from "react";

export type IExampleData<T> = T;

export interface IGenericExample<T> {
    name: string;

    component: React.ComponentClass;

    data: IExampleData<T>[];
}