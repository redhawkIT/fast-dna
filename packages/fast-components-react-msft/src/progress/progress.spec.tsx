import * as React from "react";
import * as Adapter from "enzyme-adapter-react-16";
import { configure, shallow } from "enzyme";
import examples from "./examples.data";
import { generateSnapshots } from "@microsoft/fast-jest-snapshots-react";
import { IProgressClassNameContract } from "@microsoft/fast-components-class-name-contracts-base";
import MSFTProgress, {
    IMSFTProgressClassNameContract,
    IMSFTProgressHandledProps,
    IMSFTProgressManagedClasses,
    IMSFTProgressUnhandledProps,
} from "./progress";

/*
 * Configure Enzyme
 */
configure({adapter: new Adapter()});

describe("progress snapshots", (): void => {
    generateSnapshots(examples);
});

describe("progress", (): void => {
    test("should not throw if managedClasses are not provided", () => {
        expect(
            () => {
                shallow(<MSFTProgress />);
            }
        ).not.toThrow();
    });
});
