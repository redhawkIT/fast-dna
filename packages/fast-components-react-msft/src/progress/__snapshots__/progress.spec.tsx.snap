// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`progress snapshots Progress: 0 1`] = `
<div
  aria-valuemax={100}
  aria-valuemin={0}
  aria-valuenow={50}
  className="progress-0-1-1"
  role="progressbar"
>
  <div
    className="progress_indicator-0-1-3 progress_indicator__determinate-0-1-4"
    slot="determinate"
  >
    <div
      className="progress_valueIndicator-0-1-2"
      style={
        Object {
          "width": "50%",
        }
      }
    />
  </div>
</div>
`;

exports[`progress snapshots Progress: 1 1`] = `
<div
  aria-valuemax={100}
  aria-valuemin={0}
  className="progress-0-1-1"
  role="progressbar"
>
  <div
    className="progress_indicator-0-1-3"
    slot="indeterminate"
  >
    <span
      className="progress_dot-0-1-5 progress_dot__1-0-1-6"
    />
    <span
      className="progress_dot-0-1-5 progress_dot__2-0-1-7"
    />
    <span
      className="progress_dot-0-1-5 progress_dot__3-0-1-8"
    />
    <span
      className="progress_dot-0-1-5 progress_dot__4-0-1-9"
    />
    <span
      className="progress_dot-0-1-5 progress_dot__5-0-1-10"
    />
  </div>
</div>
`;
